### PR TITLE
add 2 new text classification models

### PR DESCRIPTION
**Pull request by NonvolatileMemory**
## Description ##
add two models (region embedding ICLR 2018 and exam AAAI 2019) for text classification.

### Essentials ###
- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: scripts/text_classification/dataloader.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import numpy as np
```
Comment (eric-haibin-lin) [inline scripts/text_classification/dataloader.py]: Could you add license headers to the files, e.g. https://github.com/dmlc/gluon-nlp/blob/master/src/gluonnlp/embedding/__init__.py#L1-L19 ?

File: scripts/text_classification/region_ebmedding_readme.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# RegionEmbedding
```
Comment (eric-haibin-lin) [inline scripts/text_classification/region_ebmedding_readme.md]: Would you mind moving these instructions to the index.rst file?

File: scripts/text_classification/region_ebmedding_readme.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+- Please see the original papar about the datasets and pre-pocessing.
+- All the hyper-parameters I used are copied from the official implementation.
+## 1.Requirements
+
+- Python2 or Python3
+- Mxnet 1.2.1
+## 2.Results
+
+
+|Datasets| Accuracy(%)<br>WordContext|Best Epoch<br>WordContext|Accuracy(%)<br>ContextWord|Best Epoch<br>ContextWord|Running Time<br>Per Epoch(mins)
```
Comment (eric-haibin-lin) [inline scripts/text_classification/region_ebmedding_readme.md]: Please convert the table in rst format

File: scripts/text_classification/exam.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+base_path = 'data/'
+print_step = 200
+ctx = mx.gpu(0)
+index = 3 # which task
+n_classes = list_n_classes[index]
+vocab_size = list_vocab_size[index]
+max_sequence_length = list_max_sequence_length[index]
+task_path = list_task[index]+'/'
+####################
+class Net(nn.HybridBlock):
```
Comment (eric-haibin-lin) [inline scripts/text_classification/exam.py]: Add documentation, input/output shape

File: scripts/text_classification/exam_readme.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# AAAI_2019_EXAM
```
Comment (szha) [inline scripts/text_classification/exam_readme.md]: This should be converted to rst, and linked or inlined in https://github.com/dmlc/gluon-nlp/blob/master/scripts/text_classification/index.rst

File: scripts/text_classification/regionemb.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+net = ContextWordNet() if isContextWord else WordContextNet()
+SCE = mx.gluon.loss.SoftmaxCrossEntropyLoss()
+net.initialize(init.Xavier(), ctx=ctx)
+trainer = gluon.Trainer(net.collect_params(),'adam', {'learning_rate': learning_rate})
+data_test,data_train = load_data(data_path+task_path,max_sequence_length)
+best_acc,global_step,train_loss,train_acc = 0,0,0,0
+net.hybridize()
+ctime = time.time()
+print(ctx,list_task[index])
+for epoch in range(max_epoch):
```
Comment (szha) [inline scripts/text_classification/regionemb.py]: it would be great if the training loop could be wrapped into a function.

File: scripts/text_classification/region_ebmedding_readme.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+| :-- | :--: | :--: | :--: | :--: | :--: |
+|Yahoo Answer|73.07(73.7)|2|73.42(73.4)|3|110|
+|Amazon Polarity|95.27(95.1)|2|95.36(95.3)|3|247|
+|Amazon Full|61.58(60.9)|2|61.59(60.8)|2|183|
+|Ag news| 92.96(92.8)|6|92.89(92.8)|8|2|
+|DBPedia|98.91(98.9)|4|98.88(98.9)|3|23|
+|Yelp Full| 64.98(64.9)|3|64.94(64.5)|2|25|
+
+Note:
```
Comment (eric-haibin-lin) [inline scripts/text_classification/region_ebmedding_readme.md]: What is the exact command to run the script for training? And is the dataset expected to put in a specific folder? It's a bit hard to follow the steps here to reproduce the result fully on all these dataset

File: scripts/text_classification/dataloader.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+                print('read %d lines' % i)
+            raw = []
+            for index in slot_indexes:
+                slot_value = items[index].split()
+                tp = type_dict.get(index, int)
+                raw.append([tp(x) for x in slot_value])
+            for index in range(len(raw)):
+                slots[index].append(pad_and_trunc(raw[index],slots_lengthes[index],pad=pad,sequence=slots_lengthes[index]>1))
+    return slots
+def pad_and_trunc(data, length, pad=0, sequence=False):
```
Comment (eric-haibin-lin) [inline scripts/text_classification/dataloader.py]: do you mind at least adding an one-line comment for each of these functions?
